app.go

```diff
@@ -646,16 +689,23 @@ func (app *App) start(ctx context.Context) error {
 // called are executed. However, all those hooks are executed, even if some
 // fail.
 func (app *App) Stop(ctx context.Context) (err error) {
+
 	defer func() {
-		app.log().LogEvent(&fxevent.Stopped{Err: err})
+		// Protect the Stop hooks from being called multiple times.
+		app.runStop.Do(func() {
+			app.log().LogEvent(&fxevent.Stopped{Err: err})
+			app.closeStopChannel()
+		})
```

Comment:
Same. I'm not sure we need this guard? If there's a stop channel because of Start, then there should be a defer close.